### PR TITLE
[Minigraph-Gen]create minigraph based on testbed_vm_topology

### PR DESCRIPTION
**Pull request by maggiemsft**
### Description of PR

Generate testing minigraph.xml based on testbed topology yaml file
There is a configure_sonic_basedon_testbed.yml playbook before but only support certain topology
along with adding more testbed topologies, it cannot automatically generate minigraph based on defined testbed.yml and won't catch up with every template for each topology. 
This effort is trying to read vars/topo_*.yml and based on testbed configuration and generate matching SONiC minigraph for test run.  

### Type of change
- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### How did you do it?
1. create a generic minigraph template ansible/alltopo.j2
2. create a playbook to generate minigraph config_sonic_basedon_testbed_v2.yml 
3. will keep version_1 playbook for a while and facing them out gradually when v2 works on all platform

#### How did you verify/test it?
I tested in local DellS6000 t0, t1-lag and t1 topology and looks like working

#### Any platform specific information?
I tested in local DellS6000 t0, t1-lag and t1 topology and looks like working

#### Supported testbed topology if it's a new test case?
na

### Documentation 
https://github.com/maggiemsft/SONiC/wiki/Device-Minigraph-Generation-V2

### CODE REVIEW

File: ansible/config_sonic_basedon_testbed_v2.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  - set_fact:
+      VM_topo: "{% if 'ptf' in topo %}False{% else %}True{% endif %}"
+
+  - name: gather testbed VM informations
+    testbed_vm_info: base_vm={{ testbed_facts['vm_base'] }} topo={{ testbed_facts['topo'] }}
+    connection: local
+    when: "VM_topo | bool"
+
+  - name: find interface name mapping and indivisual interface speed if defined
+    port_alias: hwsku="{{ hwsku }}"
```
Comment (lguohan) [inline ansible/config_sonic_basedon_testbed_v2.yml]: this should be connection: local

File: ansible/templates/alltopo.j2 [hunk 2]
```diff
@@ -0,0 +1,356 @@
+<DeviceMiniGraph xmlns="Microsoft.Search.Autopilot.Evolution" xmlns:i="http://www.w3.org/2001/XMLSchema-instance">
+  <CpgDec>
```
Comment (lguohan) [inline ansible/templates/alltopo.j2]: can you sepearate cpgdec, dpgdec, MetadataDeclaration into seperate files, so they can be maintained separately?

File: ansible/config_sonic_basedon_testbed_v2.yml (hunk 1 above)
Comment (maggiemsft) [reply ansible/config_sonic_basedon_testbed_v2.yml]: changed

File: ansible/templates/alltopo.j2 (hunk 2 above)
Comment (maggiemsft) [reply ansible/templates/alltopo.j2]: done

**Comment by maggiemsft** [comment]
here is a sample for two version minigraph change for t0 topology
$/github/sonic-mgmt/ansible/minigraph$ diff str-msn2700-01.xml str-msn2700-01.xml.t0v1 
118c118
<             <a:PeersRange>10.255.0.0/25</a:PeersRange>
---
>             <a:PeersRange>10.255.0.0/27</a:PeersRange>
127c127
<             <a:PeersRange>192.168.0.0/25</a:PeersRange>
---
>             <a:PeersRange>192.168.0.0/27</a:PeersRange>
230c230
<           <Subnets>192.168.0.0/21</Subnets>
---
>           <Subnets>192.168.0.0/27</Subnets>
283,290c283,286
<           <InAcl>SNMP_ACL</InAcl>
<           <AttachTo>SNMP</AttachTo>
<           <Type>SNMP</Type>
<        </AclInterface>
<        <AclInterface>
<           <AttachTo>ERSPAN</AttachTo>
<           <InAcl>Everflow</InAcl>
<           <Type>Everflow</Type>
---
>           <AttachTo>
>           PortChannel0001;PortChannel0002;PortChannel0003;PortChannel0004;Vlan1000
>           </AttachTo>
>           <InAcl>DataAcl</InAcl>
292,295c288,292
<           <AclInterface>
<           <AttachTo>VTY_LINE</AttachTo>
<           <InAcl>ssh-only</InAcl>
<           <Type>SSH</Type>
---
>         <AclInterface>
>           <AttachTo>
>             ERSPAN
>           </AttachTo>
>           <InAcl>Everflow</InAcl>

**Pull request closed without merging**